docs2/site/docs/getting-started/field-middleware.md

```diff
@@ -48,3 +52,96 @@ _.FieldMiddleware.Use(next =>
   };
 });
 ```
+
+Also you can implement `IFieldMiddleware` interface in your Field Middleware:
+
+```csharp
+public interface IFieldMiddleware
+{
+  Task<object> Resolve(IResolveFieldContext context, FieldMiddlewareDelegate next);
+}
+```
+
+It doesn’t have to be implemented on your middleware. Then a search will be made for such a method
+with a suitable signature. Nevertheless, to improve performance, it is recommended to implement
+this interface.
+
+The middleware delegate is defined as:
+
+```csharp
+public delegate Task<object> FieldMiddlewareDelegate(IResolveFieldContext context);
+```
+
+## Field Middleware and Dependency Injection
+
+First, you are advised to read the article about [Dependency Injection](Dependency-Injection).
+
+If you use `IFieldMiddlewareBuilder.Use` overloads which accept type parameter (that is,
+those that do not accept a `IFieldMiddleware` instance or a middleware delegate) then your
+middleware creation will be delegated to DI-container. Thus, you can pass any dependencies to
+the Field Middleware constructor, provided that you have registered them correctly in DI.
+
+Also, the middleware itself should be registered in DI:
+
+```csharp
+services.AddSingleton<InstrumentFieldsMiddleware>();
+```
+
+## Known issues
+
+Perhaps the most important thing with Field Middlewares that you should be aware of is that the
+default `DocumentExecuter` applies middlewares to the schema only once before the schema is
```

Comment:
```suggestion
default `DocumentExecuter` applies middlewares to the schema only once while the schema is being
```
Just a suggestion; ok either way.  Technically it is before the `Schema.Initialize()` function runs -- but it's part of the initialization process by the document executer, and the only valid time to run it is once, before initialization.